Firestore/core/include/firebase/firestore/timestamp.h

```diff
@@ -212,6 +212,15 @@ inline bool operator==(const Timestamp& lhs, const Timestamp& rhs) {
 }
 
 #if !defined(_STLPORT_VERSION)
+
+// Make sure the header compiles even when included after `<windows.h>` without
+// `NOMINMAX` defined. `push/pop_macro` pragmas are supported by Visual Studio
+// as well as Clang and GCC.
+#pragma push_macro("min")
```

Comment:
Just to confirm, this works correctly even if there's no current definition for the macros (i.e. that undefined gets pushed on the stack and then undefined is popped back out, below)?

Reply:
Unfortunately, these are very lightly documented in both Visual Studio and GCC (and I couldn't find Clang docs at all). Experimentally, it seems to be the case: [link](https://wandbox.org/permlink/MyeekNZMaEuD4f5e).

Reply:
I played with your example and verified it works for function-like macros too: [link](https://wandbox.org/permlink/tkWNatCa8pTu6E0S).